### DISCUSSION TITLE
Looking for all pages with for loop

### DISCUSSION THREAD

**Discussion by erkin98** [jsvine/pdfplumber#381] [Q&A] [answered]
I unable to print all pages with for loop 

> import pdfplumber
> tb = []
> with pdfplumber.open(path) as pdf:
>     for i in range(len(pdf.pages)):
>         print(pdf.pages[i].extract_text())

this code only return first page

**Comment by samkit-jain** [comment, marked as answer]
Hi @erkin98 , could you confirm that the PDF indeed has more than 1 page and there is no break condition in the code that you may have missed out on adding here?

Running

```python
import pdfplumber
tb = []

with pdfplumber.open(path) as pdf:
    for i in range(len(pdf.pages)):
        print(pdf.pages[i].extract_text())
```

works fine for me. Also, instead of doing `for i in range`, you can also do

```python
for page in pdf.pages:
    print(page.extract_text())
```

since `pdf.pages` is an iterable and to get the iteration number, you can leverage using `page.page_number` (it will be 1-based and not 0-based).

If the PDF indeed has more than 1 page, request you to share the PDF and the output you are getting so that I can investigate this further.

**Reply by erkin98** [reply]
```ruby
import pdfplumber
tb = []
with pdfplumber.open(path) as pdf:
    for i in range(len(pdf.pages)):
        print(pdf.pages[i].extract_text())
```
and output is
```
<Page:1>
<Page:2>
<Page:3>
<Page:4>
```
yes my pdf has several pages i also tried 
```ruby
> for page in pdf.pages:
    print(page.extract_text())
```
same result returned

**Reply by samkit-jain** [reply]
`<Page:1>` will be printed when doing `print(page)` and not `page.extract_text()` unless the PDF also actually contains that same text. Could you please provide the PDF redacting any sensitive information from it?

**Reply by erkin98** [reply]
Update:
after first pages my pdf is scanned thank for answer
